### DISCUSSION TITLE
Hardware adeguato per HA

### DISCUSSION THREAD

**Discussion by NiKo-OnGit** [indomus/forum#2636] [GENERALE - Discussioni generiche]
Salve a tutti!
Sono nuovo di questo mondo e mi ci sto dilettando cercando di capire nel frattempo.
Volevo capire in base a cosa giudicare adeguato l'hardware su cui eseguire Home Assistant.
Al momento sto facendo girare HA su un (ex) android box TX3 su cui ho flashato armbian, e per i pochi dispositivi che attualmente controllo, 1 interruttore e qualche attuatore tuya per le aperture del cancello e delle porte dei garage ed che ho in tutto una decina di automazioni e nessuna che si esegua contemporaneamente, va più che bene, la cpu a riposo sta tra lo 0% e il 3% e non ho visto mai superare il 25%.
Ho visto che HA è installabile anche su un Raspberry Pi 3 e che dovrebbe riuscire a gestire bene una piccola casa con pochi dispositivi e qualche automazione senza troppe pretese, ma contando che viene consigliato un RasPi 4 (e che queli di nabu vendono i loro hardware basandosi proprio su quest'ultimo) evidentemente viene considerato adeguato per tutti gli usi di tutte le grandezze, ma vedo gente che sceglie dei NUC o altri mini-pc su base Celeron e 8GB ram che mi sembrano parecchio sovradimensionati per eseguirci solo HA, anche in termini di consumo elettrico (escludendo magari quelli che hanno un server casalingo in cui HA è installato insieme ad altri servizi e la differenza di consumi sarebbe irrisoria).
Contando che, sulla carta il mio box è un po' una via di mezzo tra un Raspi 3 e un 4 e prevedendo in futuro di aumentare i dispositivi installati, volevo capire fino a che punto posso spingermi con quello che ho attualmente.
Voi in base a cosa avete scelto il dispositivo?

**Comment by snakuzzo** [comment]
Ciao.
Impossibile dire dove ti puoi spingere. Le variabili sono tantissime. Posso sicuramente dirti che non ha senso cambiare se non hai un'effettiva necessità legata all'hardware poco performante. Da quello che leggo mi pare di capire che ad oggi non hai alcun problema.
Molti di quelli che comprano un NUC per HA non hanno ben chiaro di cosa hanno bisogno e comprano solo per sentito dire. Il NUC potrebbe aver senso nel caso in cui ci sia bisogno di "roba" che necessiti di tanto lavoro a livello di CPU come ad esempio il transcoding legato al flusso video delle telecamere nel caso in cui non si abbia un hardware dedicato a questo. In questo caso la soluzione più rapida per chi non ne sa niente è... "prendo la CPU più potetnte e stica".
Però, posso dirti che, fino a un annetto fa, io avevo Core e altre 10 applicazioni (tra cui anche il media server) su docker su un RPI3B...e non ho mai avuto necessità di cambiare.
Alla fine ho preso [QUESTO](https://indomus.it/componenti/mini-pc-celeron-j4125-chuwi-herobox-8gb-ram-e-256gb-ssd/) miniPC più per sfizio che per altro e ad un prezzo irrisorio.

**Reply by NiKo-OnGit** [reply]
Si, è come dici tu, al momento ho talmente poca roba da gestire che se il box non mi da problemi di altra natura (che non escluderei, visto che è comunque un prodotto cinese di manifattura non testabile) non credo di cambiarlo a breve.
Era più che altro per capire se in base alle esperienze della gente si potesse quantificare un numero di dispositivi/automazioni limite che una certo hardware riesce a gestire.
Poi se mi dici che con un RasPi 3B riuscivi a gestire tutte quelle applicazioni contemporaneamente senza problemi (anche se sicuramente il media server era eseguito senza il transcoding), vuol dire che l'uso di risorse in generale è davvero minimo

**Reply by snakuzzo** [reply]
Impossibile quantificare l'hardware in base a dispositivi e automazioni.
Sì, ovviamente è come dici, media server senza transcoding ;) ... altrimenti sarebbe morto tutto nel giro di 2 secondi :D
Le risorse che richiede Core sono veramente minime
